code/game/objects/items/candle.dm

```diff
@@ -76,3 +72,13 @@
 	if(lit)
 		return source_temperature
 	return 0
+
+/obj/item/candle/Crossed(var/obj/Proj)
+	if(istype(Proj, /obj/item/projectile/beam))
```

Comment:
Make an isBeam() macro instead of doing this manually.

Reply:
Sure, where would it live though?

Reply:
ignore me i'm stupid

Reply:
done